Interact with 3D meshes

aChanEP opened the pull request:
Description:
<!--A brief description of changes. Things to include: WIP? Dependent PR's opened against other tickets? -->

1.) Mesh does not show up when in fullscreen mode.  For some reason it disappears DONE
2.) Work on getting cursor to show up only for devices without controls DONE

This PR is part of these three tasks:

https://elasticpath.atlassian.net/secure/RapidBoard.jspa?rapidView=467&projectKey=RS&modal=detail&selectedIssue=RS-701

https://elasticpath.atlassian.net/secure/RapidBoard.jspa?rapidView=467&projectKey=RS&modal=detail&selectedIssue=RS-702

https://elasticpath.atlassian.net/secure/RapidBoard.jspa?rapidView=467&projectKey=RS&modal=detail&selectedIssue=RS-685

Also adjusted the icons for VR and AR.

Linting:
<!--Have you validated that no linting errors are introduced? -->
- [x] No linting errors

Component Updates:
<!--Have you updated components/package.json and components/package-lock.json for any components that have been updated? -->
- [x] Component package requires update on npm [@elasticpath/store-components](https://www.npmjs.com/package/@elasticpath/store-components)

Tests:
<!--Have tests been run locally and passed? If manual tests run, explain what was run below-->
- [ ] E2E tests (npm test run with `e2e`)
- [ ] Manual tests

Documentation:
<!--Are documentation updates required? Include any mention of updates required below if necessary-->
- [ ] Requires documentation updates

shaunmaharaj commented on src/components/src/ProductDisplayItem/productdisplayitem.main.tsx:
```diff
@@ -276,10 +282,12 @@ class ProductDisplayItemMain extends Component<ProductDisplayItemMainProps, Prod
       if (Config.vr.enable) {
-        vrFileExists = await ProductDisplayItemMain.urlExists(Config.vr.skuVrImagesUrl.replace('%sku%', itemLookupRes._code[0].code));
+        backgroundVRImageExists = await ProductDisplayItemMain.urlExists(Config.vr.skuVrImagesUrl.replace('%sku%', itemLookupRes._code[0].code));
+        meshVRImageExists = await ProductDisplayItemMain.urlExists(Config.vr.skuVrMeshesUrl.replace('%sku%', itemLookupRes._code[0].code));
```
excellent 👍

shaunmaharaj commented on src/components/src/VRProductDisplayItem/README.md:
```diff
@@ -12,6 +12,12 @@ For an example of how to integrate this component into your own project checkout
 
 `productdisplayitem.main.tsx`
 
+The component gives the user the ability to rotate in the vr-scene with mobile, dedicated vr devices and on desktop.  Look at the aframe website for a more exhaustive list.  
```
We should provide a link to the A-Frame website here if we're referencing it.

aChanEP replied:
Sounds good

aChanEP replied:
There is a reference above in the README.  Ill reword it to reference it more clearly

shaunmaharaj commented on src/components/src/VRProductDisplayItem/README.md:
```diff
@@ -12,6 +12,12 @@ For an example of how to integrate this component into your own project checkout
 
 `productdisplayitem.main.tsx`
 
+The component gives the user the ability to rotate in the vr-scene with mobile, dedicated vr devices and on desktop.  Look at the aframe website for a more exhaustive list.  
```
```suggestion
The component gives the user the ability to rotate in the vr-scene with mobile devices, dedicated vr devices, and on desktop.  Look at the aframe website for a more exhaustive list.  
```

shaunmaharaj commented on src/components/src/VRProductDisplayItem/README.md:
```diff
@@ -12,6 +12,12 @@ For an example of how to integrate this component into your own project checkout
+The component also allows the user to set a panorama image, a 3D mesh, and an html markup that populates a 3D plane.  
+
+Note that the html markup that populates the plane in the 3d scene
+is hidden outside of aframes `Scene` tag.  Populate markup inside the div with the classname `inner-html-texture` to customize desired information to be shown in the 3D scene.
```
Unnecessary new line.

aChanEP replied:
I've reworded it.  The newline seems to make sense to me.  Let me know if it still doesn't

shaunmaharaj commented on src/components/src/VRProductDisplayItem/README.md:
```diff
@@ -12,6 +12,12 @@ For an example of how to integrate this component into your own project checkout
+The component also allows the user to set a panorama image, a 3D mesh, and an html markup that populates a 3D plane.  
+
+Note that the html markup that populates the plane in the 3d scene
+is hidden outside of aframes `Scene` tag.  Populate markup inside the div with the classname `inner-html-texture` to customize desired information to be shown in the 3D scene.
```
Very well described!

shaunmaharaj commented on src/components/src/VRProductDisplayItem/README.md:
```diff
@@ -12,6 +12,12 @@ For an example of how to integrate this component into your own project checkout
+
+Note that the html markup that populates the plane in the 3d scene
+is hidden outside of aframes `Scene` tag.  Populate markup inside the div with the classname `inner-html-texture` to customize desired information to be shown in the 3D scene.
+
```
Remember to mention the meshURL in the example snippet.

aChanEP replied:
done

shaunmaharaj commented on src/components/src/VRProductDisplayItem/VRProductDisplayItem.less:
```diff
@@ -143,3 +146,40 @@
+}
+
+.inner-html-texture {
+    // background-image: url(cheesecake.png); 
```
leftover cheesecake 🍰

aChanEP replied:
removed

shaunmaharaj commented on src/components/src/VRProductDisplayItem/VRProductDisplayItem.less:
```diff
@@ -143,3 +146,40 @@
+}
+
+.html-text-texture {
+    background: rgb(30, 30, 30);
```
Should this color be them-able?

aChanEP replied:
Yeah it might be better to make it a seperate react component maybe?

aChanEP replied:
Made it into its own react component so someone can come in and more easily change it

shaunmaharaj commented on src/components/src/VRProductDisplayItem/VRProductDisplayItem.tsx:
```diff
@@ -66,24 +190,26 @@ class VRProductDisplayItem extends Component<IVRComponentProps, IVRComponentStat
+            </div>
+          </div>
+
+          <Scene className="vr-container" embedded vr-mode-ui="enterVRButton: #myEnterVRButton;" background="color: grey" loading-screen="backgroundColor: #000000; dotsColor: white">
```
There's a lot of inline css here. Can we move it to the less file?

aChanEP replied:
Not so sure styles will be able to attach attributes to an element.  Those are aframe specific attributes that dictate settings in the scene.

aChanEP replied:
Best I can do is maybe format it a little better?

aChanEP replied:
I just reformatted it so its easier to read

shaunmaharaj commented on src/ep.config.json:
```diff
@@ -1,7 +1,7 @@
   "cortexApi": {
     "path": "/cortex",
-    "scope": "vestri",
+    "scope": "vestri_reference",
```
Reminder to revert this 😄

aChanEP replied:
reverted

shaunmaharaj commented on src/ep.config.json:
```diff
@@ -58,7 +58,8 @@
   "vr": {
     "enable": true,
-    "skuVrImagesUrl": "https://s3.amazonaws.com/referenceexp/vr/%sku%.jpg"
+    "skuVrImagesUrl": "https://s3.amazonaws.com/referenceexp/vr/%sku%.jpg",
```
I'll add some documentation for this tomorrow 👍

aChanEP commented:
> Wonderful job with this one Alvin! Thanks for completing this 😃
> 
> Added a few minor comments. We can merge once it's set.

addressed all comments! let me know if u have any other concerns!
